### ISSUE TITLE
Copying wrapped lines in cygwin can have a spurious space/newline between lines

### ISSUE THREAD

**Issue by brendanmig** [microsoft/terminal#5760]
# Environment

```none
Platform ServicePack Version      VersionString
-------- ----------- -------      -------------
 Win32NT             10.0.18363.0 Microsoft Windows NT 10.0.18363.0

Windows Terminal
Version: 0.11.1191.0
```

# Steps to reproduce

Open a cygwin terminal and enter a long line that wraps the terminal. If you then select the entire wrapped line you see an extra space is selected where the line wraps. This does not occur in a normal cmd terminal. When you copy the long cygwin line it also copies the space, and the newline, into the clipboard, so when you paste you end up getting 2 lines with an extra space.

# Expected behavior

Expected behaviour is that the text is copied as a single line without a spurious space.

# Actual behavior

That it behave like in the cmd window. The extra space at the end of the line should not be selected. And when copied the space and newline should not be in the clipboard. This is a very annoying issue as copying and pasting commands from one terminal to another fails as it ends up pasting two lines which get run as two invalid commands.

Multiline selection in cygwin:
![image](https://user-images.githubusercontent.com/7307742/81127447-93b6b800-8f81-11ea-8ea1-e608944733d7.png)

Notice the highlighted space at the end of the first line.

Multiline selection in CMD:
![image](https://user-images.githubusercontent.com/7307742/81127466-a7621e80-8f81-11ea-95bf-f8bcd9f286b8.png)

Notice how there is no highlighted space at the end of the first line.

Pasted text from both examples in Sublime Text:
![image](https://user-images.githubusercontent.com/7307742/81127531-d6789000-8f81-11ea-9a40-b217fa07789c.png)

Note how the cygwin text is over two lines whereas the CMD isn't.

**Comment by zadjii-msft** [comment]
@brendanmig What version of cygwin are you running?

**Comment by brendanmig** [comment]
It's an old one since I don't like updating cygwin :) I guess I should update to the latest though and try again. In any case it's:
```
bendy@eric ~
$ cygcheck -V
cygcheck (cygwin) 2.10.0
System Checker for Cygwin
Copyright (C) 1998 - 2018 Cygwin Authors
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

bendy@eric ~
$ uname -a
CYGWIN_NT-10.0-WOW eric 2.10.0(0.325/5/3) 2018-02-02 15:21 i686 Cygwin
```

**Comment by DHowett-MSFT** [comment]
so, I’ve got some bad news. This might be the same as explained in https://github.com/microsoft/terminal/issues/3367#issuecomment-600331472 — the MSYS runtime and the Cygwin runtime are one and the same. This is something an update might help with, but something Terminal is totally powerless to help.

**Comment by brendanmig** [comment]
Well I just updated to cygwin64 (3.1.4) and it seems to have solved the problem. I use this as the profile:
```
            {
                // Make changes here to the cmd.exe profile
                "guid": "{0caa0dad-35be-5f56-58ff-afceeeaa6102}",
                "name": "cygwin64",
                "commandline": "C:\\cygwin64\\bin\\bash --login -i",
                "startingDirectory": "c:\\cygwin64\\bin",
                "icon": "c:\\cygwin64\\Cygwin-Terminal.ico",
                "hidden": false
            }, 
```
Which is the same as the 32 bit one except `cygwin64` instead of `cygwin`.

In a cygwin64 tab the multiline selection is fine. In a cygwin one it isn't. Maybe they fixed or updated something internally.

**Comment by DHowett-MSFT** [comment]
I believe the runtime changes in 3.1 helped fix this. Thanks for confirming!

**Comment by brendanmig** [comment]
Thanks for your fast response! This issue has been bugging me since I started using WT back in Jan, glad just a quick update could fix it.

**Comment by randersenLSI** [comment]
Hi.  I'm new to github.  I recently reinstalled cygwin on a new win10 box.  I copied all of my rc files and information from my previous version.  I am having a similar issue where if I select text from a line that has wrapped in the vim or even cat and then paste it at the cursor it is not pasting back onto one line.
Cygwin Package Information
Package              Version        Status
cygwin               3.1.6-1        OK


![mintty_7m4i6n5LgW](https://user-images.githubusercontent.com/69676928/90243934-c92f3500-ddfd-11ea-801e-0ee0ec20cfb0.png)

e.g.

![mintty_OLvm554CiW](https://user-images.githubusercontent.com/69676928/90243945-cd5b5280-ddfd-11ea-9d2b-38a72006505f.png)

My supervisor using 2.9 does not have this issue.  I tried searching for cygwin linewrap and this post is the closes to what is happening to me.  

Thanks in advance for any help.

Robert
